OpenSim/Common/Exception.h

```diff
@@ -47,7 +47,31 @@
 //=============================================================================
 //=============================================================================
 
-namespace OpenSim { 
+namespace OpenSim {
+
+
+/** Macros to throw OpenSim exceptions. The purpose of these macros is to avoid
+having to provide the first few arguments (which are common to all OpenSim
+exceptions) to the exception constructor in the throw statements. This also
+allows us to add more details (eg class name) later easily.                   
+
+Note -- Extra braces enclosing "if" are to avoid problems when these macros are
+called within if-else statements like:
+\code{.cpp}
+          if(<some condition>)
+              OPENSIM_THROW_IF(<arguments>)
+          else
+              <some statements>                                               
+\endcode                                                                      */
+#define OPENSIM_THROW_IF(CONDITION, EXCEPTION, ...)                  \
+    {                                                                \
+    if(CONDITION)                                                    \
+        throw EXCEPTION{__FILE__, __LINE__, __func__, __VA_ARGS__};  \
+    }
+
+#define OPENSIM_THROW(EXCEPTION, ...)                                \
+    OPENSIM_THROW_IF(true, EXCEPTION, __VA_ARGS__)
+
 
```

Comment:
Seems a little weird to me that `OPENSIM_THROW` calls `OPENSIM_THROW_IF(true, ...)` instead of the reverse. It seems unnecessary to involve a condition when there is none. 
```c++
+#define OPENSIM_THROW_IF(CONDITION, EXCEPTION, ...)                  \
+    {                                                                \
+    if(CONDITION)                                                    \
+        OPENSIM_THROW(EXCEPTION, __VA_ARGS__) 
+    }
+
+#define OPENSIM_THROW(EXCEPTION, ...)                                \
+     throw EXCEPTION{__FILE__, __LINE__, __func__, __VA_ARGS__};     \
```
Does that not work?